### ISSUE TITLE
Migrate some releases to helm3 leaving others to be managed by helm2

### ISSUE THREAD

**Issue by prashantkalkar** [helm/helm-2to3#149]
Hi 

I have lot of application services using helm charts to deploy into Kubernetes. 
We are also using helm charts like Prometheus operator, EFK stack etc on our prod and non prod cluster. 
We are currently using the helm2 version: `v2.14.3`

I know that helm2 and helm3 can manage the same cluster as long as the releases are different. And I can use that approach for all the stateless releases that are running on the cluster. This will allow me to migrate one release at a time, de-risking the change (mainly in production cluster). (Let me know if my assumption is wrong here).

The real issue I am facing is with **stateful** services like Prometheus and elastic search. 

**Is it possible to transfer the ownership from helm2 to helm3 of a single release, without performing the entire cluster migration?**

This will allow us to migrate the releases from helm2 to helm3 one release at a time while continuing the deployments of other releases using helm2 (we have deployment pipelines using the helm). 

If the ownership transfer is possible, then it  can even be used for stateless services as well, that way the older helm2 release will not have to be deleted. 

Please let me know if you need anymore information.

**Comment by hickeyma** [comment]
@prashantkalkar This should be possible by converting on a release by release basis. In other words, running the command `helm 2to3 convert`  on a release as you require. You can then cleanup that release from Helm v2 when you are happy the conversion is successful by running `helm 2to3 cleanup --name <release_name>`. No Kubernetes resources associated with a release are removed during conversion or cleanup.

Be aware that doing the migration to Helm v3 in a step by step basis will require you to mange your cluster, and avoid Helm v2 and Helm v3 release Kubernetes resources from clashing.

Its worth checking out the [2to3 plugin README](https://github.com/helm/helm-2to3/blob/master/README.md) and the [Helm migration doc](https://helm.sh/docs/topics/v2_v3_migration/).

**Comment by prashantkalkar** [comment]
Are you saying `helm 2to3 move config` is optional when migration is done release by release basic, allow to helm2 and helm3 releases co-exits? 

The documentation does not clarify whether it might be optional. (Its possible that might have missed it).

**Comment by hickeyma** [comment]
`helm 2to3 move config` copies Helm v2 configuration (plugins, repository list etc.) over to the new configuration set-up for Helm v3. I agree move was not a great verb to use.

The releases are separate to the configuration. `helm 2to3 convert` command does that:
https://github.com/helm/helm-2to3#readme-before-migration

Helm v2 and v3 can co-exist but need to be careful to handle not trying to make changes of same release ion both versions. If you don't cleanup releases or configuration then Helm v2 can still work.

Cleanup removes Helm v2 data, configuration and Tiller: https://github.com/helm/helm-2to3#clean-up-helm-v2-data

**Comment by prashantkalkar** [comment]
@hickeyma Thanks a lot. 

You are right, I was concern due to the word move configuration from helm2 to helm3. But now I realised that it is mainly the config setup for the helm3 and does not change the cluster configuration. 

I was able to migration some services following the steps you mentioned :
1. Backup the helm2 home directory and helm release config maps from the cluster. 
2. Move (Copy) config from helm2 to helm3 with `move` command.
3. With the `convert` command update helm3 to manage a single release. 
4. Ensure everything works and helm3 shows the service. 
5. Clean up that service with `cleanup` command with `name` option from helm2. 

I updated the pipelines and any other automation we had to ensure correct helm version is managing the services. 

This will now allow us to move one service at a time incrementally. 

Thanks a lot, closing the ticket.
